addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/HaID.java

```diff
@@ -28,10 +28,10 @@
  */
 @NonNullByDefault
 public class HaID {
-    final public String baseTopic;
-    final public String component;
-    final public String nodeID;
-    final public String objectID;
+    final private String baseTopic;
```

Comment:
a final property cannot be mutated from outside and this is even an immutable class.
There is no reason to hide the properties behind getters except if you don't like performance or readability or maintainability or want to make it extra hard for reviewers ^^.

Please make those fields public again and access them directly. The HaID class is meant to be immutable. That might be added to the javadoc for the class.